### PR TITLE
Data Type - C99 Update

### PR DESCRIPTION
**Pull request by kjwelch**
This concludes issue #99

### CODE REVIEW

File: frontend/src/lexer/lexer_core.rs
Comment (CalebLItalien) [inline frontend/src/lexer/lexer_core.rs:1]: What does are you using 'previous_char' for?

Comment (CalebLItalien) [reply frontend/src/lexer/lexer_core.rs:1]: Okay I looked through it and I see what happened now. This looks like part of it is for a different issue? Which issue is this associated with? The variants of `DataType` look like they need to be realigned with #97 still, `Boolean` and `String` are not types, and I think a few might be missing. This is on the right track though! :)

Comment (CalebLItalien) [reply frontend/src/lexer/lexer_core.rs:1]: 'double', 'long', 'signed', and 'unsigned' are the ones I'm seeing but double check the whole list as well.

**Comment by kjwelch** [comment]
Most recent push fixes the MINUS and NEGATIVE issue by implementing DASH ---> This still does not fix the parsing of unary expressions within binary expressions (which I am working on implementing now, the test at the very bottom of parser_tests is being used for this and the test prior to it tests unary expression parsing). Also, this adds 3 DataType tokens that were missed in the previous push.

**Comment by CalebLItalien** [comment]
Sounds good! Comment on this issue when you feel its ready to be reviewed again. :)

**Review by CalebLItalien — COMMENTED** [review]
Looks good! Just resolve merge conflicts and then merge it in.

**Review by CalebLItalien — APPROVED** [review]

**Pull request merged**